### PR TITLE
Improved test_subscription_options

### PR DESCRIPTION
**Pull request by ahcorde**
I runned lcov locally, some lines are missing here:

Selección_055

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

File: rclcpp/test/rclcpp/test_subscription_options.cpp [hunk 1]
```diff
@@ -85,4 +85,16 @@ TEST_F(TestSubscriptionOptions, topic_statistics_options_node_default_mode) {
...
       *(node->get_node_base_interface())));
+
+  subscription_options.topic_stats_options.state = rclcpp::TopicStatisticsState::Disable;
+  EXPECT_FALSE(
+    rclcpp::detail::resolve_enable_topic_statistics(
+      subscription_options,
+      *(node->get_node_base_interface())));
+
+  subscription_options.topic_stats_options.state = static_cast<rclcpp::TopicStatisticsState>(5);
+  EXPECT_THROW(
```
Comment (clalancette) [inline rclcpp/test/rclcpp/test_subscription_options.cpp:96]: I'd rather use `RCLCPP_EXPECT_THROW_EQ` here so we can ensure that the error message is what we expect.

**Review by clalancette — CHANGES_REQUESTED** [review]
One thing to fix.

**Review by clalancette — APPROVED** [review]
Looks good with green CI.

**Comment by ahcorde** [comment]
compiling up-to `rclcpp` and testing `rclcpp`

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12549/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7517/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10260/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12469/)

**Pull request merged**